integrationtests/tools/proxy/proxy.go

```diff
@@ -65,7 +59,15 @@ func (q *queue) Add(e packetEntry) {
 	if len(q.Packets) > 1 {
 		lastIndex := len(q.Packets) - 1
 		if q.Packets[lastIndex].Time.Before(q.Packets[lastIndex-1].Time) {
-			sort.Stable(q.Packets)
+			slices.SortStableFunc(q.Packets, func(a, b packetEntry) int {
+				if a.Time.Before(b.Time) {
+					return -1
+				}
+				if a.Time.After(b.Time) {
+					return 1
+				}
+				return 0
+			})
```

Comment:
Actually, we don't need to sort at all. This slice is always sorted. We just need to insert the new element at the correct index.